### PR TITLE
apache-arrow: enable `ARROW_S3`

### PR DESCRIPTION
**Pull request by simpl1g**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by carlocab** [comment]
Please use the preferred commit-message style for homebrew/core. We put the name of the formula first in commit-message headings.

For new formulae:

> At Homebrew, we like to put the name of the formula up front like so: `foobar 7.3 (new formula)`.

For existing formulae:

> The preferred commit message format for simple version updates is `foobar 7.3` and for fixes is `foobar: fix flibble matrix.`.

Refer to the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit) for more details. Also, when making further changes to your pull request, adhere to the following guidelines to make sure that @BrewTestBot can merge your commits:

> - One formula per commit; one commit per formula.
> - Keep merge commits out of the pull request.

File: Formula/apache-arrow.rb [hunk 1]
```diff
@@ -56,6 +58,7 @@ def install
       -DARROW_PLASMA=ON
       -DARROW_PROTOBUF_USE_SHARED=ON
       -DARROW_PYTHON=ON
+      -DARROW_S3=ON
```
Comment (carlocab) [inline Formula/apache-arrow.rb:61]: Why is it useful to enable this?

Comment (simpl1g) [reply Formula/apache-arrow.rb:61]: It allows to read/write files from AmazonS3/Min.io cloud storage. Without this option error returned for s3 related functions in apache-arrow

**Comment by simpl1g** [comment]
@carlocab Thank you for review! Should I force push commit with updated title and description?

**Comment by carlocab** [comment]
Yes please. Though, we need to get `aws-sdk-cpp` bottled on Linux first (probably in a separate PR). I'd also like to get feedback from other users of the `apache-arrow` formula, since `aws-sdk-cpp` is a pretty heavy dependency to force onto users who don't need it.

CC @jeroen @nealrichardson @ianmcook @kou @daipom

**Comment by jeroen** [comment]
Yes, we also enable this dependency in the libs that we build for the R package, e.g.

 - https://github.com/autobrew/homebrew-core/blob/master/Formula/apache-arrow.rb
 - https://github.com/autobrew/homebrew-cran/blob/master/Formula/apache-arrow-static.rb
 - https://github.com/r-windows/rtools-packages/blob/master/mingw-w64-arrow/PKGBUILD

So I think it makes sense to enable it here as well.

**Comment by carlocab** [comment]
The Linux build was skipped because `aws-sdk-cpp` has no Linux bottle. We need to sort that out first.

**Comment by kou** [comment]
@carlocab Can we merge this because you add a workaround, Homebrew/homebrew-test-bot#692 ? Or should we wait for Linux bottle of aws-sdk-cpp?

**Comment by carlocab** [comment]
That's not really a workaround. That change would make this PR fail Linux CI if we ran it again.

**Comment by simpl1g** [comment]
@carlocab can we merge this now? Ubuntu build should pass after merging this PR https://github.com/Homebrew/homebrew-core/pull/89027

**Review by carlocab — APPROVED** [review]
Thanks, @simpl1g!

**Pull request closed without merging**